### PR TITLE
Add no_push drone step and caching

### PR DESCRIPTION
**Pull request by colinhoglund**
Perform a `--no-push` build on PRs and add `/go` caching.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -16,8 +13,17 @@ steps:
       - apk add make
       - make test
 
+  - name: build
+    image: plugins/kaniko-ecr
+    pull: always
+    settings:
+      no_push: true
+    when:
+      event: [pull_request]
```
Comment (whwalter) [inline .drone.yml:22]: This would enable an unprivileged snyk scan if desired. You would have access to both the Docker file and the oci container tar.

Comment (colinhoglund) [reply .drone.yml:32]: With the caveat that we would need to expose a snyk token to PR events

File: .drone.yml [hunk 2]
```diff
@@ -29,9 +35,12 @@ steps:
         from_secret: ecr_access_key
       secret_key:
         from_secret: ecr_secret_key
+    when:
+      event: [push]
```
[the remark this answers is not in the document]
Comment (colinhoglund) [reply .drone.yml:52]: We don't really have a convention for that and all of our ECR repos have lifecycle policies, so I don't think it's that much of a concern.

File: .drone.yml (hunk 1 above)
Comment (whwalter) [reply .drone.yml:32]: This is a public repo so if you just want to snyk container test and report the build result to drone you can I believe. You would only need a token for monitoring the build in a snyk project.

Comment (colinhoglund) [reply .drone.yml:32]: It appears that all of these require auth

```
root@0106a2fb1af1:/app# snyk test
`snyk` requires an authenticated account. Please run `snyk auth` and try again.
root@0106a2fb1af1:/app# snyk container test
`snyk` requires an authenticated account. Please run `snyk auth` and try again.
```

Comment (whwalter) [reply .drone.yml:32]: I might have conflated this with just a default snyk scan then.

File: .drone.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (lviana) [reply .drone.yml:52]: I was thinking about restricting it to `main`.

File: .drone.yml (hunk 1 above)
Comment (colinhoglund) [reply .drone.yml:35]: something for a later date perhaps.

**Review by lviana — APPROVED** [review]

**Pull request merged**